### ISSUE TITLE
convergence hangs in chef-client 12.0.3, stops on access to /dev/tty. No problem in chef 11.16.

### ISSUE THREAD

**Issue by Cytrian** [chef/chef#2742]
tl;dr: On chef11, /dev/tty is not available during convergence, on chef12 it is, and programs accessing /dev/tty will hang forever.

After some hours of debugging I have simplified my problem up to this short recipe:

    execute "stop demo" do
       command "stty -tostop </dev/tty"
    end

Similar stty stuff is done in "/usr/bin/resize" in package "xterm" on debian, or in "zookeeper-client", so it's nothing exotic.

Target platform is debian wheezy.

On chef 11 I get a message

    ------------------------------------
    Expected process to exit with [0], but received '2'
    ---- Begin output of stty -tostop </dev/tty ----
    STDOUT: 
    STDERR: sh: 1: cannot open /dev/tty: No such device or address
    ---- End output of stty -tostop </dev/tty ----
    Ran stty -tostop </dev/tty returned 2

That's okay, as this is usally catched.

In chef 12, /dev/tty seems to be available, stty is stopped by SIGTTOU and the overall convergence hangs forever. stty tries to do some output and is not allowed to do so.

A more elaborated example to show this problem is the following recipe:

    package "xterm"

    execute "stop demo" do
        command "echo -e '1\n2\n3' | parallel --gnu --progress -X -N1 -j20 'echo X{}X' >/tmp/output.txt"
    end

**Comment by jekriske** [comment]
I believe this issue is related to https://github.com/chef/chef/issues/3189 
Lamont: you're the expert, any progress on this?

**Comment by lamont-granquist** [comment]
This is a dup, fixed in mixlib-shellout
